pkg/new-ui/v1beta1/authzn.go

```diff
@@ -0,0 +1,106 @@
+package v1beta1
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/kubeflow/katib/pkg/util/v1beta1/env"
+	v1 "k8s.io/api/authorization/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ENV variables
+var (
+	USER_HEADER  = env.GetEnvOrDefault("USERID_HEADER", "kubeflow-userid")
+	USER_PREFIX  = env.GetEnvOrDefault("USERID_PREFIX", ":")
+	DISABLE_AUTH = env.GetEnvOrDefault("APP_DISABLE_AUTH", "false")
+)
+
+func GetUsername(r *http.Request) (string, error) {
+	var username string
+	if DISABLE_AUTH == "true" {
+		log.Printf("APP_DISABLE_AUTH set to True. Skipping authentication check")
+		return "", nil
+	}
```

Comment:
Can we avoid duplicate check for [`if DISABLE_AUTH == "true"`](https://github.com/kubeflow/katib/blob/9e283983cb26a4bb318d2d86d7795f3b2b4bca2e/pkg/new-ui/v1beta1/authzn.go#L64-L68) by adding `GetUsername` logic under `IsAuthorized` call ?
Then, we have 1 place where we verify if auth is required.
WDYT @apo-ger ?

Reply:
Sure, I moved the `GetUsername` logic under `IsAuthorized` and introduced the appropriate checks to handle the various errors properly!